### PR TITLE
Protocol packages utils restructuring

### PR DESCRIPTION
**Pull request by amir-blox**
This PR changes the packages containing utilities, test mocks, etc.

- `testing` dir that exports testing utilities from some  package (added for `ssv`, `qbft`)
- added mock-p2p-network with messaging

### CODE REVIEW

File: operator/validator/controller.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"crypto/rsa"
 	"encoding/hex"
+	commons2 "github.com/bloxapp/ssv/protocol/v2/qbft"
```
Comment (nkryuchkov) [inline operator/validator/controller.go:7]: Could you please rename it from `commons2` to something else? E. g. `qbftprotocol`

File: protocol/v2/message/consensus_test.go [hunk 2]
```diff
@@ -4,7 +4,7 @@ import (
 	"sort"
 	"testing"
 
-	testing2 "github.com/bloxapp/ssv/protocol/v2/testing"
+	qbfttesting "github.com/bloxapp/ssv/protocol/v2/testing"
```
Comment (nkryuchkov) [inline protocol/v2/message/consensus_test.go:7]: Why is it `qbfttesting`, not `ssvtesting`?

File: protocol/v2/p2p/testing/utils.go [hunk 3]
```diff
@@ -0,0 +1,408 @@
+package testing
+
+import (
+	"context"
+	crand "crypto/rand"
```
Comment (nkryuchkov) [inline protocol/v2/p2p/testing/utils.go:5]: Why `crand`?

File: protocol/v2/p2p/testing/utils.go [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+	"encoding/hex"
+	"encoding/json"
+	"github.com/bloxapp/ssv/protocol/v2/message"
+	protocolp2p "github.com/bloxapp/ssv/protocol/v2/p2p"
```
Comment (nkryuchkov) [inline protocol/v2/p2p/testing/utils.go:9]: I think we need to decide at this point what names we use for importing packages inside `protocol`: If `protocol` is a prefix or a postfix. I used it as a postfix in a few places, but probably having it as a prefix is better, also we already have a `spec` prefix.

@lior-blox @nivBlox Your thoughts?

File: protocol/v2/p2p/testing/utils.go [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+		if err := m.SendStreamMessage("last_changeround", pi, msg); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+//
+//func (m *mockNetwork) GetHistory(mid spectypes.MessageID, from, to specqbft.Height, targets ...string) ([]SyncResult, specqbft.Height, error) {
```
Comment (nkryuchkov) [inline protocol/v2/p2p/testing/utils.go:322]: Do we need to keep it?

File: protocol/v2/p2p/testing/utils.go (hunk 3 above)
Comment (amir-blox) [reply protocol/v2/p2p/testing/utils.go:5]: to differentiate from `math/rand`

File: protocol/v2/message/consensus_test.go (hunk 2 above)
Comment (amir-blox) [reply protocol/v2/message/consensus_test.go:7]: I'll change to `protocoltesting`

**Review by nkryuchkov — APPROVED** [review]

**Pull request merged**